openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java

```diff
@@ -157,12 +148,25 @@ private Optional<String> retrieveValueForAnnotationFields(AnnotationNode service
                 .map(en -> en.toString().trim());
     }
 
-    private String generateOpenApiDoc(SemanticModel semanticModel, SyntaxTree syntaxTree,
-                                      ServiceDeclarationNode serviceNode, String outputFileName) {
-        ModulePartNode modulePartNode = syntaxTree.rootNode();
+    private void generateOpenApiDoc(OpenApiDocConfig config, SyntaxNodeAnalysisContext context, NodeLocation location) {
+        int serviceId = config.getSemanticModel().hashCode();
+        String targetFile = String.format(FILE_NAME_FORMAT, serviceId);
+        ModulePartNode modulePartNode = config.getSyntaxTree().rootNode();
         List<ListenerDeclarationNode> listenerNodes = extractListenerNodes(modulePartNode);
-        return ServiceToOpenAPIConverterUtils.generateOASForGivenFormat(serviceNode, true, listenerNodes,
-                semanticModel, outputFileName);
+        OASResult oasResult = ServiceToOpenAPIConverterUtils.generateOAS(
+                config.getServiceNode(), listenerNodes, config.getSemanticModel(), targetFile, null);
```

Comment:
Is it impossible to take the ballerina file path here, if we can have a path, then we can resolve the contract path which binds to openapi annotation and update the given contract with title and version 🙂 ?

Reply:
We can take the ballerina file path here. But, we are currently not interested in title and version of the OpenAPI definition. Since, there is an open issue [1] for this will plan this migration in SL Update 3.

[1] - [https://github.com/ballerina-platform/ballerina-standard-library/issues/2420](https://github.com/ballerina-platform/ballerina-standard-library/issues/2420)